In [1]:
import numpy as np
import scipy.stats as ss

# Задача 1

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр $17$ мм. Используя уровень значимости $0.025$ и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из $100$ шариков средний диаметр оказался равным $17.5$ мм, а дисперсия известна и равна $4$ мм.

In [2]:
a = 17
alpha = 0.025
n = 100
mean = 17.5
sigma = np.sqrt(4)

In [3]:
T = (mean - a) / (sigma / np.sqrt(n))
T

2.5

Учитывая, что статистика $T$ имеет стандартное распределение и $\alpha = 0.025$, т.е. это квантиль порядка $0.975$, то правосторонняя критическая область - $(2, {\infty})$. В таком случае, $T = 2.5$ входит в критическую область и гипотеза $H_0$ должна быть отвергнута.

# Задача 2

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением $16$. Найти доверительный интервал для оценки математического ожидания $a$ с надёжностью $0.95$, если выборочное среднее равно $80$, а объём равен $256$.

In [4]:
sigma = 16
alpha = 0.95
mean = 80
n = 256

Так как среднее квадратическое $\sigma$ распределения известна, то $t$ высчитывается для нормального распределения. 

In [5]:
t = ss.norm.ppf(1-alpha / 2)
t

0.06270677794321385

In [6]:
a = (mean - t * sigma / np.sqrt(n),mean + t * sigma / np.sqrt(n))
a

(79.93729322205678, 80.06270677794322)

Доверительный интервал для матожидания $a$ в таком случае будет $(79.9373, 80.0627)$

# Задача 3

Продавец утверждает, что средний вес пачки печенья составляет $200$ г. Из партии извлечена выборка из $10$ пачек. Вес каждой пачки составляет:

$202, 203, 199, 197, 195, 201, 200, 204, 194, 190$

Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен $1\%$?

In [7]:
a = 200
n = 10
X_ = np.array([202,203,199,197,195,201,200,204,194,190])
alpha = 0.01

Так как $\sigma$ нормального распределения не известна, найдем несмещенную оценку среднего квадратического и среднее значение по выборке

In [8]:
sigma_x = X_.std(ddof=1)
mean = X_.mean()

Статистика $T$ в таком случае высчитывается с учетом $\sigma_X$

In [9]:
T = (mean - a) / (sigma_x / np.sqrt(n))
T

-1.0651074037450896

Так как статистика $T$ имеет распределение $t$-стьюдента, критическая область для него посчитаем с помощью `t.ppf` и $df = n - 1$

In [10]:
t = ss.t.ppf(1 - alpha/2, df = n - 1)
t

3.2498355440153697

Критическая область в таком случае будет $(\infty, -3.2498) \cup(3.2498, \infty)$

Теперь сравним значение статистики $T$ с критической областью

In [11]:
if T > -t and T < t:
    print('Утверждение продавца верно.')
else:
    print('Продавец врет.')

Утверждение продавца верно.
